tests/integration/cmd_link_unlink_test.go

```diff
@@ -9,138 +13,203 @@ import (
-				helper.CopyExample(filepath.Join("source", "nodejs"), frontendContext)
-				helper.Cmd("odo", "create", "--s2i", "nodejs", "frontend", "--context", frontendContext, "--project", commonVar.Project).ShouldPass()
-				helper.Cmd("odo", "push", "--context", frontendContext).ShouldPass()
-				helper.CopyExample(filepath.Join("source", "python"), backendContext)
-				helper.Cmd("odo", "create", "--s2i", "python", "backend", "--context", backendContext, "--project", commonVar.Project).ShouldPass()
-				helper.Cmd("odo", "push", "--context", backendContext).ShouldPass()
-				stdErr := helper.Cmd("odo", "link", "backend", "--context", frontendContext, "--port", "1234").ShouldFail().Err()
-				Expect(stdErr).To(ContainSubstring("Unable to properly link to component backend using port 1234"))
-			})
+	When("two components are deployed", func() {
+		var frontendContext, backendContext, frontendURL, frontendComp, backendComp string
+
+		// checkDescribe: checks that the linked component and related variables are present in the output of odo describe
+		var checkDescribe = func(contextDir string, compName string, pushed bool, bindAsFiles bool) {
+			stdOut := helper.Cmd("odo", "describe", "--context", contextDir).ShouldPass().Out()
+			Expect(stdOut).To(ContainSubstring("Linked Services"))
+			Expect(stdOut).To(ContainSubstring(compName))
+			if pushed {
+				if bindAsFiles {
+					Expect(stdOut).To(ContainSubstring("/bindings"))
+				} else {
+					Expect(stdOut).To(ContainSubstring("SERVICE_BACKEND_IP"))
+					Expect(stdOut).To(ContainSubstring("SERVICE_BACKEND_PORT"))
+				}
+			}
+		}
+
+		// createAndPush: creates component, a URL for it and deploys it
+		var createAndPush = func(compType string, compName string, contextDir string) {
+			helper.CopyExample(filepath.Join("source", compType), contextDir)
+			helper.Cmd("odo", "create", compType, compName, "--context", contextDir, "--project", commonVar.Project).ShouldPass()
+			helper.Cmd("odo", "url", "create", "--port", "8080", "--context", contextDir).ShouldPass()
+			helper.Cmd("odo", "push", "--context", contextDir).ShouldPass()
+		}
+
+		BeforeEach(func() {
+			frontendComp = fmt.Sprintf("frontend-%v", helper.RandString(3))
+			frontendContext = helper.CreateNewContext()
+			createAndPush("nodejs", frontendComp, frontendContext)
+			frontendURL = helper.DetermineRouteURL(frontendContext)
+
+			backendComp = fmt.Sprintf("backend-%v", helper.RandString(3))
+			backendContext = helper.CreateNewContext()
+			createAndPush("python", backendComp, backendContext)
 		})
-	*/
 
-	/*
-		Context("When handling link/unlink between components", func() {
-			JustBeforeEach(func() {
-				frontendContext = helper.CreateNewContext()
-				backendContext = helper.CreateNewContext()
+		AfterEach(func() {
+			helper.DeleteDir(frontendContext)
+			helper.DeleteDir(backendContext)
+		})
+
+		When("a link is created between the two components", func() {
+			BeforeEach(func() {
+				// we link
```

Comment:
I'd much rather remove the comment because the `When` block-string seems self-explanatory to me, Idk why I added that comment, but maybe in another PR?